experimental/dds/tree/src/test/utilities/PendingLocalStateTests.ts

```diff
@@ -153,5 +91,82 @@ export function runPendingLocalStateTests(
 			expect(unuploadedEditChunks.length).to.equal(0);
 			expect(editLog.getEditLogSummary().editChunks.length).to.equal(2);
 		});
+
+		it('applies and submits ops from 0.0.2 in 0.0.2', async () =>
+			applyStashedOp(WriteFormat.v0_0_2, WriteFormat.v0_0_2));
+		it('applies and submits ops from 0.0.2 in 0.1.1', async () =>
+			applyStashedOp(WriteFormat.v0_0_2, WriteFormat.v0_1_1));
+		it('applies and submits ops from 0.1.1 in 0.0.2 (via upgrade)', async () =>
+			applyStashedOp(WriteFormat.v0_1_1, WriteFormat.v0_0_2));
+
+		it('applies and submits ops from 0.1.1 in 0.1.1', async () =>
+			applyStashedOp(WriteFormat.v0_1_1, WriteFormat.v0_1_1));
+
+		async function applyStashedOp(treeVersion: WriteFormat, opVersion: WriteFormat): Promise<void> {
+			const {
+				container: stashingContainer,
+				tree: stashingTree,
+				testObjectProvider,
+			} = await setUpLocalServerTestSharedTree({
+				id: documentId,
+				writeFormat: opVersion,
+			});
+			const stashingTestTree = setUpTestTree(stashingTree);
+			const { tree: observerTree } = await setUpLocalServerTestSharedTree({
+				id: documentId,
+				testObjectProvider,
+				writeFormat: treeVersion,
+			});
+			const url = (await stashingContainer.getAbsoluteUrl('/')) ?? fail('Container unable to resolve "/".');
+			await testObjectProvider.ensureSynchronized();
+			const initialEditLogLength = stashingTree.edits.length;
+
+			const { pendingLocalState, actionReturn: edit } = await withContainerOffline(
+				testObjectProvider,
+				stashingContainer,
+				() =>
+					stashingTree.applyEdit(
```

Comment:
We should have a test that includes an override, whether here or as a UT in IdCompressor.
Also, thoughts on testing non-insert ops?

Reply:
Since the IdCompressor API has not changed anymore, I added an override to the test here.